### DISCUSSION TITLE
fyne package -ldflags "-w -s -H=windowsgui" doesn't reduce executable size

### DISCUSSION THREAD

**Discussion by tsc-pygy** [fyne-io/fyne#5161] [Q&A] [answered]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

Hello,

Thanks for your great work on the fyne project. Since the sample code on the fyne website (to create one dialog with one button) built into a 30 MB executable I followed the information here: https://github.com/fyne-io/fyne/issues/2393. I tried doing this command as suggested: package build -ldflags "-w -s -H=windowsgui" and it seemed to build, but the executable size didn't change. Is there a way to fix this?

I'm using linux. Thanks for any assistance.

### How to reproduce

Just read what I did above, I guess.

### Example code

package build -ldflags "-w -s -H=windowsgui"

package main

import (
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/container"
	"fyne.io/fyne/v2/widget"
)

func main() {
	a := app.New()
	w := a.NewWindow("Hello")

	hello := widget.NewLabel("Hello Fyne!")
	w.SetContent(container.NewVBox(
		hello,
		widget.NewButton("Hi!", func() {
			hello.SetText("Welcome :)")
		}),
	))

	w.ShowAndRun()
}

### Fyne version

2.5.1

### Go compiler version

1.23.1

### Operating system and version

Linux Mint 20.1

### Additional Information

Sorry, I wasn't sure what you wanted for sample code.

**Comment by Jacalz** [comment, marked as answer]
Coming from the fact that you are using Linux, I would guess that the Windows specific linker option `-H=windowsgui` causes the compiler to abort because it isn't supported on Linux. Are you sure that it actually built a new binary?

PS: Please use Discussions for questions. The issue tracker is for bugs and feature requests.

**Comment by andydotxyz** [comment]
FYI using "fyne package -release" will provide an optimised package using the correct flags for the target operating system.

**Comment by tsc-pygy** [comment]
Thanks for the info,

When I tried "fyne package -release" the executable was larger (around 22 MB) than when I tried "fyne package tags -ldflags -w -s" (a little under 15 MB). Removing "H=windowsgui" did the trick, I guess. Thanks again.
